### DISCUSSION TITLE
Improved syntax for faults (backwards compatible)

### DISCUSSION THREAD

**Discussion by fmontesi** [jolie/jolie#442] [Ideas (Features, Fixes, etc.)]
A little idea for making the syntax for throwing and handling faults a bit more intuitive and in line with other tech.

The following modifications are *additions*: the old syntax would still work.

1. Introduce `throw FaultName`, equivalent to `throw( FaultName )`.
2. Introduce `throw FaultName( data )`, equivalent to `throw( FaultName, data )`.
3. Introduce `install( FaultName( x ) => P )`. This handles FaultName and puts the contained data in `x`, instead of `scopeName.FaultName`.

1 and 2 should be straightforward improvements and are easy to implement. Not sure about 3 though.

Any comments? :thinking:

**Comment by fmontesi** [comment]
Forgot to add: I'd start by implementing 1 and 2, then leave 3 for a longer discussion.

**Comment by klag** [comment]
ok for 1 and 2

about 3: I think it is useful. I would expand better the case of default that is very common to be used. In that case it is often necessary to know the name of the fault, so maybe the x should contain not only the data of the fault but also some extra meta data about the fault

**Reply by fmontesi** [reply]
Yeah I was thinking about it and it might look like this

```
install( default( faultName, data ) => ... )
```

but I'd like to think a bit more about this syntax. It'd be nice to find something where data remains the only parameter, to match the throw statement.

**Comment by fmontesi** [comment]
Now implemented (parts 1 and 2). :)
